packages/doc-contribs/pr-antora-content-guidelines-checker/src/steps/PageFilenameStep.ts

```diff
@@ -0,0 +1,93 @@
+import * as core from "@actions/core";
+import {
+  ActionResult,
+  ValidationResult,
+  Status,
+  ValidationStep,
+} from "../validation";
+import { getFileContent } from "actions-common";
+import { GitHub } from "@actions/github/lib/utils";
+
+export class PageFilenameStep extends ValidationStep {
+  name: string;
+  description: string;
+  stepResult: ActionResult | null;
+  files: string[];
+
+  constructor(files: string[], extensionsToCheck: string[]) {
+    super();
+    this.name = "Page filename";
+    this.description =
+      "Some filenames are not following the kebab-case convention. To ensure consistency and maintain best practices, please rename the following files accordingly:";
+    this.stepResult = null;
+    this.files = this.filterFiles(files, extensionsToCheck);
+  }
+
+  formatCommentBody(): string {
+    if (!this.stepResult || this.stepResult.status === Status.SUCCESS) {
+      core.debug(`No section for ${this.name} step will be write.`);
+      return "";
+    }
+
+    let commentBody = `## 🥙 ${this.name} \n`;
+    commentBody += `${this.description}\n`;
+    this.stepResult.results.forEach((actionResult) => {
+      commentBody += `- [ ] Rename **${actionResult.file}** to **${actionResult.details}**\n`;
+    });
+    commentBody += `
+> [!warning]
+> Additionally, don't forget to add the \`:page-aliases:\` attribute after renaming the **existing files** to avoid broken links or references. `;
```

Comment:
**suggestion:** Here the warning applies to file that previously existed prior the PR creation.
If the file is a new file created in the PR, there is no need to add an alias (and it shouldn't be added).
Let's talk in person to find a more convenient message.